tests/pages/base_page.py

```
@@ -177,67 +234,176 @@ def select_from_autocomplete(
+
+        except StaleElementReferenceException:
+
+            options = visible_options(
+                self.driver
+            )
+
+            if not options:
+                raise TimeoutException(
+                    "Autocomplete options "
+                    "became stale"
+                )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)